I forgot to mention in my review just now that both RTEMS-4 CI builds of this PR are failing with the same compiler errors, I think this is the right command-line:
```
  make[4]: Entering directory '/home/runner/work/epics-base/epics-base/modules/libcom/src/O.RTEMS-pc386-qemu'
  In file included from ../misc/cantProceed.h:28,
  /opt/rtems/4.9/bin/i386-rtems4.9-g++ -B/opt/rtems/4.9/i386-rtems4.9/pc386/lib/ -specs bsp_specs -qrtems    -mtune=i386               -DUNIX      -O2 -g -g   -Wall         -D__LINUX_ERRNO_EXTENSIONS__ -DRTEMS_LEGACY_STACK   -I. -I../O.Common -I. -I../osi/compiler/gcc -I../osi/compiler/default -I. -I../osi/os/RTEMS-score -I../osi/os/RTEMS -I../osi/os/posix -I../osi/os/default -I.. -I../as -I../bucketLib -I../calc -I../cvtFast -I../cppStd -I../cxxTemplates -I../dbmf -I../ellLib -I../env -I../error -I../fdmgr -I../flex -I../freeList -I../gpHash -I../iocsh -I../log -I../macLib -I../misc -I../osi -I../pool -I../ring -I../taskwd -I../timer -I../yacc -I../yacc -I../yajl -I../../../../include/compiler/gcc -I../../../../include/os/RTEMS -I../../../../include         -c ../osi/os/RTEMS-score/osdTime.cpp
                   from ../osi/epicsMutex.h:44,
                   from ../osi/epicsThread.h:365,
                   from ../timer/epicsTimer.h:21,
                   from ../osi/generalTimeSup.h:15,
                   from ../osi/os/RTEMS-score/osdTime.cpp:29:
  /opt/rtems/4.9/lib/gcc/i386-rtems4.9/4.3.2/../../../../i386-rtems4.9/include/stdlib.h:80:25: error: macro "free" requires 2 arguments, but only 1 given
  /opt/rtems/4.9/lib/gcc/i386-rtems4.9/4.3.2/../../../../i386-rtems4.9/include/stdlib.h:87:35: error: macro "malloc" requires 3 arguments, but only 1 given
  In file included from ../misc/cantProceed.h:28,
                   from ../osi/epicsMutex.h:44,
                   from ../osi/epicsThread.h:365,
                   from ../timer/epicsTimer.h:21,
                   from ../osi/generalTimeSup.h:15,
                   from ../osi/os/RTEMS-score/osdTime.cpp:29:
  /opt/rtems/4.9/lib/gcc/i386-rtems4.9/4.3.2/../../../../i386-rtems4.9/include/stdlib.h:80: error: variable or field 'free' declared void
  ../osi/os/RTEMS-score/osdTime.cpp:172: warning: 'done' defined but not used
  make[4]: *** [../../../../configure/RULES_BUILD:268: osdTime.o] Error 1
  make[4]: *** Waiting for unfinished jobs....
```

I have no idea what's causing those failures, but they do need to be fixed.

--- a/modules/libcom/src/misc/cantProceed.c
+++ b/modules/libcom/src/misc/cantProceed.c
@@ -15,6 +15,7 @@
 #include <stdarg.h>
 
 #include "errlog.h"
+#include "envDefs.h"
 #include "cantProceed.h"
 #include "epicsThread.h"
 #include "epicsStackTrace.h"
@@ -53,20 +54,30 @@ LIBCOM_API void * mallocMustSucceed(size_t size, const char *msg)
 
 LIBCOM_API void cantProceed(const char *msg, ...)
 {
+    int shouldAbort = 0;
     va_list pvar;
     va_start(pvar, msg);
     if (msg)
         errlogVprintf(msg, pvar);
     va_end(pvar);
 
-    errlogPrintf(ANSI_RED("CRITICAL ERROR") " Thread %s (%p) can't proceed, suspending.\n",
-            epicsThreadGetNameSelf(), epicsThreadGetIdSelf());
+    envGetBoolConfigParam(&EPICS_ABORT_ON_ASSERT, &shouldAbort);
+
+    errlogPrintf(ANSI_RED("CRITICAL ERROR") " Thread %s (%p) can't proceed, %s.\n",
+            epicsThreadGetNameSelf(), epicsThreadGetIdSelf(), shouldAbort ? "aborting" : "suspending");
 
     epicsStackTrace();
 
     errlogFlush();
 
     epicsThreadSleep(1.0);
-    while (1)
-        epicsThreadSuspendSelf();
+
+    if (shouldAbort) {
+        errlogPrintf("cantProceed() Calling abort()\n");
+        errlogFlush();
+        abort();
+    } else {
+        while (1)
+            epicsThreadSuspendSelf();
+    }
 }

--- a/modules/libcom/src/misc/cantProceed.h
+++ b/modules/libcom/src/misc/cantProceed.h
@@ -34,12 +34,14 @@
 extern "C" {
 #endif
 
-/** \brief Suspend this thread, the caller cannot continue or return.
+/** \brief Suspend this thread or call abort(), the caller cannot continue
+ *         or return.
  *
  * The effect of calling this is to print the error message followed by
- * the name of the thread that is being suspended. A stack trace will
+ * the name of the thread that can't proceed. A stack trace will
  * also be shown if supported by the OS, and the thread is suspended
- * inside an infinite loop.
+ * inside an infinite loop, or the process aborted, depending on
+ * EPICS_ABORT_ON_ASSERT.
  * \param errorMessage A printf-style error message describing the error.
  * \param ... Any parameters required for the error message.
  */

--- a/modules/libcom/src/osi/epicsMutex.h
+++ b/modules/libcom/src/osi/epicsMutex.h
@@ -213,12 +213,12 @@ LIBCOM_API epicsMutexLockStatus epicsStdCall epicsMutexLock(
  * This routine does not return if the identifier is invalid.
  * \param ID The mutex identifier.
  **/
-#define epicsMutexMustLock(ID) {                            \
-    epicsMutexLockStatus status = epicsMutexLock(ID);       \
-    if(status != epicsMutexLockOK) {                        \
-        cantProceed("epicsMutexMustLock() failed at %s:%d", \
-            __FILE__, __LINE__);                            \
-    }                                                       \
+#define epicsMutexMustLock(ID) {                                \
+    epicsMutexLockStatus status = epicsMutexLock(ID);           \
+    if(status != epicsMutexLockOK) {                            \
+        cantProceed("epicsMutexMustLock() failed at %s:%d\n",   \
+            __FILE__, __LINE__);                                \
+    }                                                           \
 }
 
 /**\brief Similar to epicsMutexLock() except that the call returns immediately,

--- a/modules/libcom/src/osi/os/RTEMS-score/osdTime.cpp
+++ b/modules/libcom/src/osi/os/RTEMS-score/osdTime.cpp
@@ -15,18 +15,19 @@
 
 #define EPICS_EXPOSE_LIBCOM_MONOTONIC_PRIVATE
 #include <epicsStdio.h>
-#include <rtems.h>
 #include <errno.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
-#include <rtems/rtems_bsdnet_internal.h>
 #include "epicsTime.h"
 #include "osdTime.h"
 #include "osiNTPTime.h"
 #include "osiClockTime.h"
 #include "generalTimeSup.h"
+
+#include <rtems.h>
+#include <rtems/rtems_bsdnet_internal.h>
 
 extern "C" {
 

